app/models/reference.rb

```diff
@@ -29,4 +29,9 @@ def email_address_not_own
 
     errors.add(:email_address, :own) if email_address == candidate_email_address
   end
+
+  def update_token
+    raw_token, _encrypted_token = Devise.token_generator.generate(Reference, :token)
+    update(token: raw_token)
```

Comment:
I would've expected the `raw_token` to be sent to the user, and only the encrypted token saved in the database. 

This is how `VendorApiToken` and `Candidate` tokens work - it prevents the tokens from being used if they ever leak. Open to discussion here!

Reply:
Yes, I should have added a big `TODO` there, it needs to be secured.

Reply:
Acctually we added that iten to the https://trello.com/c/B4tokVQU/§, because we were aware this PR was getting long.

Reply:
Done